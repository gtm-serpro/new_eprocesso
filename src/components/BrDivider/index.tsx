import classNames from "classnames";
import React from "react";
import { useMtProps } from "../../hooks/useMtProps";
import { useSpreadProps } from "../../hooks/useSpreadProps";
import IMtProps from "../../types/IMtProps";

export interface BrDividerProps extends React.HTMLAttributes<HTMLDivElement>, IMtProps {
  /** Se o divider será vertical ou não. */
  vertical?: boolean;
  /** Se é tracejado ou não. */
  dashed?: boolean;
  /** Tamanho.
   *
   * - sm: 2x o tamanho original
   * - md: 3x o tamanho original
   * - lg: 4x o tamanho original
   */
  size?: string;
}

const BrDivider = React.forwardRef<HTMLDivElement, BrDividerProps>(
  ({ className, children, vertical = false, dashed = false, size, ...props }, ref) => {
    const mtProps = useMtProps(props);
    const spreadProps = useSpreadProps(props);

    return (
      <div
        ref={ref}
        className={classNames(
          "br-divider",
          vertical && "vertical",
          dashed && "dashed",
          size === "sm" && "sm",
          size === "md" && "md",
          size === "lg" && "lg",
          ...mtProps,
          className
        )}
        {...spreadProps}
      >
        {children}
      </div>
    );
  }
);

export default BrDivider;
