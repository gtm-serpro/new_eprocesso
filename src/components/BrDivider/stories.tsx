import { Meta, Story } from "@storybook/react";

import BrDivider, { BrDividerProps } from ".";

export default {
  title: "Utilidades/Divider",
  component: BrDivider,
} as Meta;

export const Horizontal: Story<BrDividerProps> = () => (
  <div>
    <p>Lorem ipsum dolor sit amet consectetur adipisicing elit.</p>
    <BrDivider mt={3} mb={3} />
    <p>Lorem ipsum dolor sit, amet consectetur adipisicing elit.</p>
  </div>
);

export const Vertical: Story<BrDividerProps> = () => (
  <div className="d-flex">
    <p>Lorem ipsum dolor sit amet consectetur adipisicing elit.</p>
    <BrDivider ml={3} mr={3} vertical={true} />
    <p>Lorem ipsum dolor sit, amet consectetur adipisicing elit.</p>
  </div>
);
