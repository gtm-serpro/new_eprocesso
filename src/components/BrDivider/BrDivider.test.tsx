import { render } from "@testing-library/react";

import BrDivider from ".";

describe("<BrDivider />", () => {
  it("deveria renderizar o divider simples", () => {
    const { container } = render(<BrDivider />);
    expect(container.firstChild).toHaveClass("br-divider");
    expect(container.firstChild).toMatchSnapshot();
  });

  it("deveria renderizar o divider vertical se prop informada", () => {
    const { container } = render(<BrDivider vertical />);
    expect(container.firstChild).toHaveClass("vertical");
  });

  it("deveria renderizar o divider simples com margin se prop informada", () => {
    const { container } = render(<BrDivider m={3} />);
    expect(container.firstChild).toHaveClass("m-3");
  });
});
