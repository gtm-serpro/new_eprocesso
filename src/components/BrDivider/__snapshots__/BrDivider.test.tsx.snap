// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrDivider /> deveria renderizar o divider simples 1`] = `
<div
  class="br-divider"
/>
`;
